unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Font/Providers/AddressableFontProvider.cs

```diff
@@ -0,0 +1,21 @@
+using Cysharp.Threading.Tasks;
+using System.Threading;
+using TMPro;
+using UnityEngine;
+using UnityEngine.AddressableAssets;
+
+namespace DCL.Providers
+{
+    public class AddressableFontProvider : IFontAssetProvider
+    {
+        private readonly AddressableResourceProvider addressableProvider;
+
+        public AddressableFontProvider(AddressableResourceProvider provider)
```

Comment:
Use `IAddressableResourceProvider` instead, otherwise `IAddressableResourceProvider` is really redundant. And in a proper DI you should rely on abstractions anyway.

Moreover in this test
```
[Test]
        public async Task RetrievedCorrectlyAnAddressableFont()
        {
            var fontName = "NotoSansJP-SemiBold SDF";
            var fallbackFontToLoad = $"Assets/Fonts/Fonts Sources/{fontName}.asset";

            var addressableFontProvider = new AddressableFontProvider(new AddressableResourceProvider());

            TMP_FontAsset asset = await addressableFontProvider.GetFontAsync(fallbackFontToLoad);

            Assert.NotNull(asset);
            Assert.AreEqual(fontName, asset.name);
        }
```
if you had some complex logic behind `AddressableResourceProvider ` you would have to mock it, and it is not possible with the final class